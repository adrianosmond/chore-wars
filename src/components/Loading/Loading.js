import React from 'react';

import './Loading.css';

import loading1 from './loading1.png';
import loading2 from './loading2.png';

const Loading = () => (
  <div className="loading">
    <div className="loading__logo">
      <div className="loading__image-wrapper">
        <img src={loading1} alt="" className="loading__image loading__image--1" />
        <img src={loading2} alt="" className="loading__image loading__image--2" />
      </div>
      <h1 className="loading__text">Loading</h1>
    </div>
  </div>
);

export default Loading;
