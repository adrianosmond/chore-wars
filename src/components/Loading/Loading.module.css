.outer {
  display: flex;
  flex-direction: column;
  align-items: center;
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -100px;
  margin-left: -66px;
}

.inner {
  position: relative;
  width: 59px;
  height: 119px;
  transform-origin: 50% 100%;
  animation: wobble 2s infinite;
}

.image {
  position: absolute;
  top: 0;
  left: 0;
}

.image2 {
  top: 20px;
  left: 4px;
  transform-origin: 110% -10%;
  animation: pulltrigger 0.5s infinite;
}

@keyframes pulltrigger {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(-15deg);
  }
}

@keyframes wobble {
  0% {
    transform: rotate(-5deg);
  }

  50% {
    transform: rotate(5deg);
  }

  100% {
    transform: rotate(-5deg);
  }
}
