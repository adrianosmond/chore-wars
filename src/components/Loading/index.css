.loading {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
}

.loading__logo {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.loading__image-wrapper {
  position: relative;
  width: 59px;
  height: 119px;
  transform-origin: 50% 100%;
  animation: wobble 2s infinite;
}

.loading__image {
  position: absolute;
  top: 0;
  left: 0;
}

.loading__image--2 {
  top: 20px;
  left: 4px;
  transform-origin: 110% -10%;
  animation: pulltrigger 0.5s infinite;
}

.loading__text {
  animation: wobble 1.5s infinite;
}

@keyframes pulltrigger {
  0% {
    transform: rotate(0deg)
  }

  100% {
    transform: rotate(-15deg);
  }
}

@keyframes wobble {
  0% {
    transform: rotate(-5deg)
  }

  50% {
    transform: rotate(5deg);
  }

  100% {
    transform: rotate(-5deg)
  }
}
