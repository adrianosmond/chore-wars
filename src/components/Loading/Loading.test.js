import React from 'react';
import { shallow } from 'enzyme';
import Loading from './Loading';

describe('Loading', () => {
  it('Renders', () => {
    expect(shallow(<Loading />)).toMatchSnapshot();
  });
});
