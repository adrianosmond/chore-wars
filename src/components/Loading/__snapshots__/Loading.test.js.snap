// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Renders 1`] = `
<div
  className="loading"
>
  <div
    className="loading__image-wrapper"
  >
    <img
      alt=""
      className="loading__image loading__image--1"
      src="loading1.png"
    />
    <img
      alt=""
      className="loading__image loading__image--2"
      src="loading2.png"
    />
  </div>
  <h1
    className="loading__text"
  >
    Loading
  </h1>
</div>
`;
